simulations/plotting.py

```diff
@@ -40,7 +41,60 @@ def add_arrow(line, size=20, color='k'):
 
     x = np.cos(theta)
     y = ellipt * np.sin(theta)
-    (xr, yr) = np.matmul(rotmat, [x.flatten(1), y.flatten(1)])
+    (xr, yr) = np.matmul(rotmat, [x.flatten("C"), y.flatten("C")])
+
+    if axes == None:
+        plt.figure(figsize=[size, size])
+        axes = plt.axes()
+    else:
+        plt.sca(axes)
+    axes.axis('square')
+    line = plt.plot(xr, yr)[0]
+    add_arrow(line)
+    plt.xlim(-1, 1)
+    plt.ylim(-1, 1)
+    
+    
+def plotEllipse2(S1, S2, S3, S4, size=5, axes=None):
```

Comment:
* the Stokes parameters are named S0, S1, S2, S3 (https://en.wikipedia.org/wiki/Stokes_parameters).  * it is better to give plotEllipse2 a more descriptive name, e.g., plotEllipseStokes. 
```suggestion
def plotEllipseStokes(S0, S1, S2, S3, size=5, axes=None):
```
* Notebook sets S0 to integer and the other three Stokes parameters to float. They are all float. The maximum value of S1, S2, S3 can at most be S0, since S0 represents total intensity of light.